### DISCUSSION TITLE
Versioning system for Models / Pages / Workflows (and others?)

### DISCUSSION THREAD

**Discussion by JFQueralt** [cortezaproject/corteza-server#823] [Ideas]
Rationale: It would make a lot of sense to have some sort of versioning system for a number of objects in the platform such as
- Models
- Pages
- Workflows
- Settings

Case scenarios:
- Being able to undo a Module, Page, Workflow or set of Settings to a previous state easily.
- Share those differentials from version to version via export.

Further comments: To keep it short (further rationale can be given if requested), I see potential to connect Corteza with a Git, say Github, to keep some transactions reflected on a repo in the form of Issues. Such approach could actually take care of this.

**Comment by French-Asgeir** [comment]
Upvoted !

I would also find this feature extremely useful.

Regarding workflow, it might be useful to have the workflow revision logged in the sessions admin view.
